holoviews/plotting/bokeh/element.py

```diff
@@ -1405,8 +1409,15 @@ def _get_colormapper(self, eldim, element, ranges, style, factors=None, colors=N
 
         ncolors = None if factors is None else len(factors)
         if eldim:
-            if dim_name in ranges:
+            # check if there's an actual value (not np.nan)
+            if util.isfinite(self.clim).all():
+                low, high = self.clim
+            elif dim_name in ranges:
                 low, high = ranges[dim_name]['combined']
+                # Pad zero-range to avoid breaking colorbar
+                if low == high:
+                    low -= low / 10
```

Comment:
This logic won't work if low and high are zero, maybe use the default_span instead:

```python
offset = self.default_span/2
```